applications/AdjointFluidApplication/custom_utilities/numerical_diffusion.h

```diff
@@ -0,0 +1,407 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 license: KratosAdjointFluidApplication/license.txt
+//
+//  Main author:    Suneth Warnakulasriya, https://github.com/sunethwarna
+//
+
+
+#if !defined(KRATOS_NUMERICAL_DIFFUSION)
+#define KRATOS_NUMERICAL_DIFFUSION
+
+// System includes
+#include <vector>
+#include <string>
+
+// External includes
+
+// Project includes
+#include "includes/model_part.h"
+#include "includes/process_info.h"
+#include "includes/model_part.h"
+#include "includes/kratos_parameters.h"
+#include "utilities/openmp_utils.h"
+
+// Eigen library includes
+#include <Eigen/SVD>
+#include <Eigen/Eigenvalues>
+
+
+// Application includes
+#include "custom_elements/vms_adjoint_element.h"
+
+namespace Kratos
+{
+///@addtogroup AdjointFluidApplication
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// A Numerical Diffusion Calculation class.
+class NumericalDiffusion
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    KRATOS_CLASS_POINTER_DEFINITION(NumericalDiffusion);
+
+    typedef Element::IndexType IndexType;
+
+    typedef Element::SizeType SizeType;
+
+    typedef Element::MatrixType MatrixType;
+    
+    typedef Element::GeometryType GeometryType;
+
+    typedef Element::VectorType VectorType;
+
+    enum class NumericalDiffusionMethod
+    {
+        singularValuePressureCoupled,
+        singularValuePressureDecoupled,
+        eigenValueFullMatrix
+    };
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+    void SetNumericalDiffusionParameters(Parameters DiffusionParameters)
+    {
+        KRATOS_TRY;
+
+        Parameters default_params(R"(
+        {
+            "method": "singular_value_pressure_coupled",
+            "beta": 0.0
+        })");
+
+        DiffusionParameters.ValidateAndAssignDefaults(default_params);
+
+        mBeta = DiffusionParameters["beta"].GetDouble();
+
+        std::string method_name = DiffusionParameters["method"].GetString();
+        
+        if (method_name.compare("singular_value_pressure_coupled")==0)
```

Comment:
strings can be compared with `==` which is more readable imo

Reply:
done